### DISCUSSION TITLE
CPR Fails to Work at Medical Skill Level 100

### DISCUSSION THREAD

**Discussion by Agandara205** [FakeFishGames/Barotrauma#15329] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I've encountered a bug where a character with a medical skill level of 100 is unable to perform CPR effectively. The animation plays, but it has no actual effect. However, when another character with a lower medical skill performs CPR on the same target, it works as intended.

This issue occurred in my playthrough with some mods enabled. To investigate further, I disabled all mods and started a standard single-player game. I spawned 10 experience books to level up the medical skill to 100 and tested the scenario again. I injured a character until they were knocked out and attempted to perform CPR, but it still had no effect. Switching to another character with a lower medical skill allowed CPR to work correctly.

This bug seems to be reproducible and independent of mod interference.

### Reproduction steps

Start a game and disable all mods.
Level up a character's medical skill to 100 using experience books.
Injure another character until they are knocked out.
Attempt to perform CPR with the level 100 medical skill character.
Observe that the CPR animation plays but has no effect or sound.
Perform CPR with a different character with a lower medical skill and observe that it works correctly.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Couldn't reproduce with your steps

There's easier way, try this:
- go to sub editor, start test on any sub
- f3, `setskill medical 100`
- switch to the victim bot, `giveaffliction "Oxygen low" 150`
- perform CPR

> Injure another character

Are you sure you injured him with oxygen low? CPR only heals oxygen low

> The animation plays, but it has no actual effect

CPR is random, even on 100lvl, there's like 9 dummy pushes and then 1 healing push, mb you just got unlucky

**Reply by Regalis11** [reply]
I gave this a test as well, and saw similar results: CPR seems to work with no issues at medical skill level 100.

**Reply by Agandara205** [reply]
Thank you for the quick assistance! I followed your advice and set up the scenario again to test if it worked as expected. You were absolutely right performing CPR works regardless of the skill level. I may have been mistaken in thinking it didn’t work because something peculiar does persist: the "sounds" that normally accompany CPR seem to disappear when using a character with a skill level of 100. I might have confused this absence of sound with the action not functioning, as these sounds are still present when using a lower-skill character.

Once again, thank you for your help and prompt response! 🙏

**Reply by SomeRandomNoobKekeke** [reply]
> the "sounds" that normally accompany CPR seem to disappear when using a character with a skill

Ye, i think that's the sound of causing internal damage to the patient when your skill is too low
![barodev small](https://github.com/user-attachments/assets/9533ae14-6a6b-4c00-9cd6-7005385c0a01)
